testworld/park_test.go

```diff
@@ -46,3 +46,70 @@ func TestHost_Happy(t *testing.T) {
 	getInvoiceAndCheck(eCharlie, docIdentifier, "GBP")
 	fmt.Println("Host test success")
 }
+
+func TestHost_AddExternalCollaborator(t *testing.T) {
+	alice := doctorFord.getHost("Alice")
+	bob := doctorFord.getHost("Bob")
+	charlie := doctorFord.getHost("Charlie")
+	eAlice := alice.createHttpExpectation(t)
+	eBob := bob.createHttpExpectation(t)
+	eCharlie := charlie.createHttpExpectation(t)
+
+	a, err := alice.id()
+	if err != nil {
+		t.Error(err)
+	}
+
+	b, err := bob.id()
+	if err != nil {
+		t.Error(err)
+	}
+
+	c, err := charlie.id()
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Alice shares invoice document with Bob first
+	res, err := alice.createInvoice(eAlice, map[string]interface{}{
+		"data": map[string]interface{}{
+			"invoice_number": "12324",
+			"due_date":       "2018-09-26T23:12:37.902198664Z",
+			"gross_amount":   "40",
+			"currency":       "GBP",
+			"net_amount":     "40",
+		},
+		"collaborators": []string{b.String()},
+	})
+	if err != nil {
+		t.Error(err)
+	}
+	docIdentifier := res.Value("header").Path("$.document_id").String().NotEmpty().Raw()
+	if docIdentifier == "" {
+		t.Error("docIdentifier empty")
+	}
+	getInvoiceAndCheck(eAlice, docIdentifier, "GBP")
```

Comment:
Can we pass in a map of keys and values instead of just currency here?

Reply:
I agree, I will add that to be more flexible in the future.